src/InfiniteArrays.jl

```diff
@@ -161,10 +161,10 @@ axistype(::OneToInf{V}, ::OneTo{T}) where {T,V} = OneToInf{promote_type(T,V)}()
 # returns the range of indices of v equal to x
 # if v does not contain x, returns a 0-length range
 # indicating the insertion point of x
-function searchsorted(v::AbstractVector, x, ilo::Int, ::PosInfinity, o::Ordering)
+function searchsorted(v::AbstractVector, x, ilo::Integer, ::PosInfinity, o::Ordering)
     lo = ilo-1
     hi = ℵ₀
-    @inbounds while lo < hi-1
```

Comment:
Why were the inbounds removed?

Reply:
The access may not be within bounds, unless we consider this to be an internal method that is never directly called by a user. Otherwise, a call like
```julia
julia> v = [1,2]
2-element Vector{Int64}:
 1
 2

julia> searchsorted(v, 2, firstindex(v), ∞, Base.Order.ForwardOrdering())
ERROR: BoundsError: attempt to access 2-element Vector{Int64} at index [1000]
```
may lead to memory corruption through out-of-bounds access.

The alternative is to add a bounds-check to the function, in which case we may retain the inbounds annotations.

Reply:
Hmm I copied this code from `Base` which has the exact same issue:
```julia
julia> searchsorted([1,2], 2, 1, 10, Base.Order.ForwardOrdering())
2:3
```
I've started an issue: https://github.com/JuliaLang/julia/issues/51176

I think its best to just check that `hi ≤ length(v)` here and keep the inbounds

Reply:
I have reverted the inbounds for now. This is unlikely to be an issue in practice, as these are internal methods. We may revisit this in the future, if necessary.